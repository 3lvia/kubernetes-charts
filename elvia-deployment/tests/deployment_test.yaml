suite: test deployment
templates:
  - deployment.yaml
tests:
  - it: deployment snapshot should match
    set:
      image.tag: latest
      name: demo-api
      namespace: core
      microserviceType: webapi
      environment: dev
      replicaCount: 1
      resources:
        requests:
          cpu: 100m
          memory: 128Mi
        limits:
          cpu: 400m
          memory: 500Mi
    asserts:
      - matchSnapshot: {}
