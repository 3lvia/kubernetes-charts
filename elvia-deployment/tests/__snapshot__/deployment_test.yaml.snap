deployment snapshot should match:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      name: demo-api
      namespace: core
    spec:
      replicas: 1
      selector:
        matchLabels:
          app: demo-api
          microservice-type: webapi
      strategy:
        rollingUpdate:
          maxSurge: 1
          maxUnavailable: 0
        type: RollingUpdate
      template:
        metadata:
          annotations:
            kubectl.kubernetes.io/default-container: demo-api
          labels:
            app: demo-api
            azure.workload.identity/use: "true"
            microservice-type: webapi
        spec:
          containers:
            - env:
                - name: ELVIA_APPLICATION_NAMESPACE
                  value: core
                - name: ELVIA_APPLICATION_NAME
                  value: demo-api
                - name: ELVIA_ENVIRONMENT
                  value: dev
                - name: ELVIA_WORKLOAD_IDENTITY_NAME
                  value: uai-core-demo-api-dev
                - name: OTEL_EXPORTER_OTLP_PROTOCOL
                  value: grpc
                - name: OTEL_EXPORTER_OTLP_ENDPOINT
                  value: http://otel-collector.monitoring.svc.cluster.local:4317
              image: containerregistryelvia.azurecr.io/core-demo-api:latest
              imagePullPolicy: Always
              name: demo-api
              resources:
                limits:
                  cpu: 400m
                  memory: 500Mi
                requests:
                  cpu: 50m
                  memory: 100Mi
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
                runAsNonRoot: true
          imagePullSecrets:
            - name: containerregistryelvia
          securityContext:
            fsGroup: 1001
            runAsGroup: 1001
            runAsUser: 1001
            seccompProfile:
              type: RuntimeDefault
            supplementalGroups:
              - 1001
          serviceAccountName: demo-api
